test: test_fun_args

ze42 opened the pull request:
Validate we properly have fun arguments in job cache
- for direct minion
- for a minion called via syndic

### What does this PR do?
Just one more test.

Note: from my test, it fails on `v2018.3.3`, works on `v2018.3.4`

Including the test ensure we do not get regression on that part.

### What issues does this PR fix or reference?

Partialy related to #45823

The result is still not present in the mentioned event, but at least the job cache seems to have the proper load info.

### Tests written?
Yes. It is only a test.

### Commits signed with GPG?

Yes

Commit:
test: test_fun_args

Validate we properly have fun arguments in job cache
- for direct minion
- for a minion called via syndic

dwoz commented:
@ze42 This needs to be opened against the `master` branch, not `develop`. Sorry for the confusion, I realize the contributor docs are currently out of date. We've updated the docs but that change won't get deployed until we release at the end of the month.

ze42 commented:
@dwoz  ok, thanks for the info. Rebased, and changed the PR to reflect that.
(might not have checked the doc, just remembered what I did last time! :)

dhiltonp committed:
Merge branch 'master' into test_fun_args

dwoz committed:
Merge branch 'master' into test_fun_args

dwoz committed:
Blacken changed files

sagetherage committed:
Merge branch 'master' into test_fun_args

sagetherage committed:
Merge branch 'master' into test_fun_args

garethgreenaway committed:
Merge branch 'master' into test_fun_args

dwoz commented:
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.
